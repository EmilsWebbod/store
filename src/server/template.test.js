import Template from './template';

let wrapper;
let props;

describe('The Template component', () => {
  beforeEach(() => {
    props = {
      manifestJSBundle: 'manifest.js',
      mainJSBundle: 'main.js',
      vendorJSBundle: 'vendor.js',
      mainCSSBundle: 'main.css',
      root: '<div>Empty</div>',
      initialState: {},
    };
    wrapper = Template(props);
  });

  it('should return a string to Express', () => {
    expect(typeof wrapper).toBe('string');
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle not passing bundles in the props gracefully', () => {
    props = {
      root: '<div>Empty</div>',
      initialState: {},
    };
    wrapper = Template(props);
    expect(wrapper).toMatchSnapshot();
  });
});
