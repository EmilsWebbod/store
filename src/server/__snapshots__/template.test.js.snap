// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Template component should handle not passing bundles in the props gracefully 1`] = `
"
    <!DOCTYPE html>
    <html lang=\\"en\\"  >
      <head>
        <title data-react-helmet=\\"true\\"></title>
        <meta httpEquiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\" />
        <meta charSet='utf-8' />
        <meta httpEquiv=\\"Content-Language\\" content=\\"en\\" />
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\" />
        <meta name=\\"theme-color\\" content=\\"white\\" />
        
        
        <link id=\\"favicon\\" rel=\\"shortcut icon\\" href=\\"/favicon.png\\" sizes=\\"16x16 32x32\\" type=\\"image/png\\" />
        <link rel=\\"manifest\\" href=\\"manifest.json\\">
        
      </head>
      <body  >
        <div id=\\"root\\" class=\\"root\\"><div><div>Empty</div></div></div>
        <script>
          window.__PRELOADED_STATE__ = [object Object]
        </script>
        
      
      
      
    </body>
  </html>
  "
`;

exports[`The Template component should match the snapshot 1`] = `
"
    <!DOCTYPE html>
    <html lang=\\"en\\"  >
      <head>
        <title data-react-helmet=\\"true\\"></title>
        <meta httpEquiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\" />
        <meta charSet='utf-8' />
        <meta httpEquiv=\\"Content-Language\\" content=\\"en\\" />
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\" />
        <meta name=\\"theme-color\\" content=\\"white\\" />
        
        
        <link id=\\"favicon\\" rel=\\"shortcut icon\\" href=\\"/favicon.png\\" sizes=\\"16x16 32x32\\" type=\\"image/png\\" />
        <link rel=\\"manifest\\" href=\\"manifest.json\\">
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"main.css\\">
      </head>
      <body  >
        <div id=\\"root\\" class=\\"root\\"><div><div>Empty</div></div></div>
        <script>
          window.__PRELOADED_STATE__ = [object Object]
        </script>
        
      <script defer src=\\"manifest.js\\"></script>
      <script defer src=\\"vendor.js\\"></script>
      <script defer src=\\"main.js\\"></script>
    </body>
  </html>
  "
`;
